fix(test): FORBIDDEN_KEYS substring false-positive on answer_format_compliance in test_no_forbidden_keys_in_output

## Summary

`tests/test_eval_by_format_aggregate_regression.py::TestExtractAggregateByFormat::test_no_forbidden_keys_in_output` fails on every PR since PR #650 was merged.

## Repro

```
EMBEDDING_BACKEND=hashing .venv/bin/python -m pytest \
  tests/test_eval_by_format_aggregate_regression.py::TestExtractAggregateByFormat::test_no_forbidden_keys_in_output -v
```

```
AssertionError: 'answer' unexpectedly found in
"{'by_format': {'hwp': ...}, 'answer_format_compliance': 1.0, ...}"
Forbidden key 'answer' leaked into aggregate output
```

## Root Cause

`FORBIDDEN_KEYS` (`scripts/run_real_eval_delta.py:219`) contains the bare string `"answer"`. The test at line 212–217 checks:

```python
self.assertNotIn(forbidden, str(out), ...)
```

This is a **substring check** on `str(out)` (the dict's string representation), not an exact key check. PR #650 legitimately added `answer_format_compliance` as a top-level metric in `extract_aggregate`'s output dict. The substring `"answer"` is present in `"answer_format_compliance"`, so the guard fires as a false positive.

The intent of `FORBIDDEN_KEYS` is to prevent per-case payload bleed (the ADR 0003 `answer` dict). The metric key `answer_format_compliance` is safe and intended — the test logic is wrong.

## Fix Direction

Replace `assertNotIn(forbidden, str(out))` (substring match) with an exact dict-key check, e.g. recursively collect all keys from `out` and assert `forbidden not in all_keys`. Alternatively, verify only top-level keys if that's sufficient for the guard's intent.

This is a **test bug**, not a production code bug. `extract_aggregate` is correct.

## Affected Files

- `tests/test_eval_by_format_aggregate_regression.py` (line 212–217)
- `scripts/run_real_eval_delta.py` (FORBIDDEN_KEYS definition context)

## Notes

- Not caused by any recent PR — the mismatch was introduced when `answer_format_compliance` was added in PR #650 but the test's substring check wasn't updated.
- Does NOT require a `schema_version` bump or ADR.

--- a/tests/test_eval_by_format_aggregate_regression.py
+++ b/tests/test_eval_by_format_aggregate_regression.py
@@ -209,10 +209,19 @@ class TestExtractAggregateByFormat(unittest.TestCase):
             {"hwp": {"num_predictions": 1, "accuracy": 1.0, "case_results": "should_be_dropped"}}
         )
         out = extract_aggregate(summary)
+
+        def _all_keys(d: dict) -> set:
+            keys = set(d.keys())
+            for v in d.values():
+                if isinstance(v, dict):
+                    keys |= _all_keys(v)
+            return keys
+
+        present = _all_keys(out)
         for forbidden in FORBIDDEN_KEYS:
             self.assertNotIn(
                 forbidden,
-                str(out),
+                present,
                 f"Forbidden key '{forbidden}' leaked into aggregate output",
             )
 

